starter-code/src/functions-and-arrays.js

```diff
@@ -1,16 +1,57 @@
 // Iteration #1: Find the maximum
+const maxOfTwoNumbers = (arg1, arg2) => (arg1 > arg2 ? arg1 : arg2);
+maxOfTwoNumbers(1,2);
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
+const findLongestWord = wordArray => {
+	let wordLength = "";
+	wordArray.forEach(function(word) {
+		wordLength += word.length;
+  });
+  // the ... in Math.max allows an array (or in this case a string) as input. It's called Spread I believe...
+	result = wordLength.indexOf(Math.max(...wordLength));
+	return result >= 0 ? wordArray[result] : null;
+};
+
+//console.log(findLongestWord(words));
+
+findLongestWord(words);
+
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+const sumArray = arg1 => {
+  let calculateSum = 0;
+  arg1.forEach(function(numArr) {
+	  calculateSum += numArr;
+  });
+  return calculateSum;
+}
+
+console.log(sumArray(numbers));
+
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+const averageNumbers = arg1 => {
+	let amountOfItems = 0;
+	let calculateSum = 0;
+	for (let i = 0; i < arg1.length; i++) {
+		amountOfItems = i + 1;
+	}
+	arg1.forEach(function(numArr) {
+		calculateSum += numArr;
+	});
+	return amountOfItems > 0 ? calculateSum / amountOfItems : null;
```

Comment:
You can also use 'arg1.length', if it's less than one then return null and otherwise return calculateSum/arg1.length